nixos/modules/services/web-apps/nextcloud.nix

```diff
@@ -79,6 +80,59 @@ in {
       default = "/var/lib/nextcloud";
       description = "Storage path of nextcloud.";
     };
+    datadir = mkOption {
+      type = types.nullOr types.str;
+      default = null;
```

Comment:
It would be slightly more elegant to set this to `cfg.home` by default, then you can get rid of the `if/else` at the top.

However, please do so by declaring `defaultText` in the option and `datadir = mkOptionDefault cfg.home` in the config-block. That way one doesn't have to rebuild the manual if this option gets changed.